dxr/app.py

```diff
@@ -22,12 +23,14 @@
                     es_alias_or_not_found)
 from dxr.exceptions import BadTerm
 from dxr.filters import FILE, LINE
-from dxr.lines import html_line
+from dxr.lines import (html_line, tags_per_line, triples_from_es_refs,
+                       triples_from_es_regions, finished_tags)
 from dxr.mime import icon, is_image
-from dxr.plugins import plugins_named
+from dxr.plugins import plugins_named, all_plugins
 from dxr.query import Query, filter_menu_items
 from dxr.utils import (non_negative_int, decode_es_datetime, DXR_BLUEPRINT,
-                       format_number)
+                       format_number, append_update, append_by_line, cumulative_sum)
+from dxr.vcs import VcsTree
```

Comment:
I'm still not thrilled with this name. VcsTree sounds like it's a tree of VCSs. Looking at the docstring and methods for inspiration, what about VcsCache? That's what it is. Or VcsAssigner. That's what it does. Or VcsMapper. Or, if all we need from it is really a map-like interface (someVcsCache['some/path']), we could even implement `__getitem__` and call it VcsMap (though that makes the callsites a little less obvious).